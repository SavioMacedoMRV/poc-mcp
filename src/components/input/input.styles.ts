import styled from 'styled-components'
import {cores} from 'resources/cores'
import {fontes} from 'resources/fontes'
import {margens} from 'resources/margens'
import {FC, HTMLAttributes, SVGProps} from 'react'
import {BotaoIcone} from 'components/botaoIcone'
import {TextoProps} from 'components/texto/texto.styles'
import {IconeProps} from 'components/icone'
import {hexToRgba} from 'helpers/cores'

export interface IconeInput extends Omit<IconeProps, 'icone'> {
  icone?: FC<SVGProps<SVGSVGElement>>
}

export interface InputProps extends HTMLAttributes<HTMLInputElement> {
  valor?: string | number
  cor?: string
  altura?: number
  largura?: number
  tipo?: 'pesquisa' | 'numero' | 'textarea'
  numeroLinhas?: number
  numeroMaxLinhas?: number
  itemSelecionado?: boolean
  placeholder?: string
  corBorda?: string
  corBordaFocada?: string
  semBorda?: boolean
  textoProps?: TextoProps
  icone?: IconeInput
  desabilitaIcone?: boolean
  desabilitado?: boolean
  corTextoDesativado?: string
  iconeEsquerda?: boolean
  aoSubmeter?: (valor?: string | number) => void
  aoDesfocar?: (valor?: string | number) => void
  aoDigitar?: (valor?: string | number) => void
  aoFocar?: () => void
  'data-testid'?: string
}
interface StyledProps {
  $semBorda?: boolean
  $corBorda?: string
  $corBordaFocada?: string
  $desabilitado?: boolean
  $corTextoDesativado?: string
  $iconeEsquerda?: boolean
  $tipoTextArea?: boolean
  $numeroMaxLinhas?: number
}

export const InputContainer = styled.div<InputProps & StyledProps>`
  display: flex;
  padding: ${({$tipoTextArea}) =>
    $tipoTextArea ? `${margens.xxsmall}px 0 ` : `${margens.xsmall}px`};
  gap: ${margens.xsmall}px;
  flex-direction: ${({$iconeEsquerda}) =>
    $iconeEsquerda ? 'row-reverse' : 'row'};
  justify-content: ${({$iconeEsquerda}) =>
    $iconeEsquerda ? 'flex-end' : 'space-between'};
  height: ${({altura, $tipoTextArea}) => {
    if ($tipoTextArea) {
      return 'auto'
    }
    if (altura) {
      return `${altura}px`
    }
    return 'auto'
  }};
  width: ${({largura}) => (largura ? `${largura}px` : 'auto')};
  background: ${({$desabilitado}) =>
    $desabilitado ? cores.background01 : cores.neutralXXLight};
  border-radius: ${margens.xsmall}px;
  border: ${({$semBorda, $corBorda, $desabilitado}) => {
    if ($semBorda) {
      return 'none'
    }
    if ($desabilitado) {
      return `2px solid ${cores.background01}`
    }

    return `2px solid ${$corBorda ?? cores.neutralDark}`
  }};

  &:focus-within {
    border: ${({$semBorda, $corBordaFocada, $corBorda}) => {
      if ($semBorda) {
        return 'none'
      }
      if ($corBordaFocada) {
        return `2px solid ${$corBordaFocada}`
      }
      return `2px solid ${$corBorda ?? cores.neutralDark}`
    }};
    box-shadow: ${({$semBorda}) =>
      $semBorda
        ? 'none'
        : `0px 2px 8px 0px ${hexToRgba(cores.neutralXDark, 0.3)}`};
  }
`

export const InputStyled = styled.input<InputProps & TextoProps & StyledProps>`
  border: none;
  outline: none;
  background: transparent;
  width: 100%;
  font-size: ${({tamanho}) => tamanho ?? 14}px;
  line-height: ${({alturalinha}) =>
    alturalinha ? alturalinha + 'px' : 'normal'};
  font-family: ${({estilo}) => (estilo ? fontes[estilo] : fontes.semibold)};
  color: ${({cor}) => cor ?? cores.neutralXXDark};
  text-align: ${({alinhamento}) => alinhamento ?? 'start'};
  cursor: ${({cursor}) => cursor ?? 'text'};
  text-transform: ${({transformar}) => transformar ?? 'none'};

  &:focus {
    &::placeholder {
      color: transparent;
    }
  }

  &:disabled {
    color: ${({$corTextoDesativado}) =>
      $corTextoDesativado ?? cores.neutralMedium};
    cursor: ${({cursor}) => cursor ?? 'default'};
  }

  &::-webkit-search-cancel-button {
    -webkit-appearance: none;
    background: ${cores.neutralMedium};
    height: 10px;
    width: 10px;
    border-radius: 5px;
  }

  &::-webkit-inner-spin-button {
    -webkit-appearance: none;
  }
  &::-webkit-outer-spin-button {
    -webkit-appearance: none;
  }
`

export const TextAreaStyled = styled.textarea<
  InputProps & TextoProps & StyledProps
>`
  border: none;
  outline: none;
  background: transparent;
  width: 100%;
  font-size: ${({tamanho}) => tamanho ?? 14}px;
  line-height: ${({alturalinha}) =>
    alturalinha ? alturalinha + 'px' : 'normal'};
  font-family: ${({estilo}) => (estilo ? fontes[estilo] : fontes.semibold)};
  color: ${({cor}) => cor ?? cores.neutralXXDark};
  text-align: ${({alinhamento}) => alinhamento ?? 'start'};
  cursor: ${({cursor}) => cursor ?? 'text'};
  text-transform: ${({transformar}) => transformar ?? 'none'};
  resize: none;
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  white-space: normal;
  box-sizing: border-box;

  &:focus {
    &::placeholder {
      color: transparent;
    }
  }

  &:disabled {
    color: ${({$corTextoDesativado}) =>
      $corTextoDesativado ?? cores.neutralMedium};
    cursor: ${({cursor}) => cursor ?? 'default'};
  }

  &::-webkit-search-cancel-button {
    -webkit-appearance: none;
    background: ${cores.neutralMedium};
    height: 10px;
    width: 10px;
    border-radius: 5px;
  }

  &::-webkit-inner-spin-button {
    -webkit-appearance: none;
  }
  &::-webkit-outer-spin-button {
    -webkit-appearance: none;
  }
`

export const InputIconePesquisa = styled(BotaoIcone).attrs({
  semMargem: true,
})`
  align-items: baseline;
`
