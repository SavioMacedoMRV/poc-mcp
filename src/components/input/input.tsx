import React, {useState, useEffect, useRef, useCallback} from 'react'
import {StyleSheetManager} from 'styled-components'
import {
  InputStyled,
  InputProps,
  InputContainer,
  InputIconePesquisa,
  TextAreaStyled,
  IconeInput,
} from './input.styles'
import PesquisarSVG from 'icons/iPesquisar.svg'
import FecharSVG from 'icons/iFechar.svg'
import {Icone} from 'components/icone'

export const Input = ({
  tipo,
  valor,
  cor,
  corBorda,
  corBordaFocada,
  corTextoDesativado,
  itemSelecionado,
  numeroLinhas = 1,
  numeroMaxLinhas,
  placeholder,
  semBorda,
  textoProps,
  icone,
  desabilitaIcone,
  desabilitado,
  iconeEsquerda,
  'data-testid': testId = 'input',
  aoDesfocar,
  aoFocar,
  aoSubmeter,
  aoDigitar,
  ...props
}: Readonly<InputProps>) => {
  const tipoPesquisa = tipo === 'pesquisa'
  const tipoNumero = tipo === 'numero'
  const tipoTextArea = tipo === 'textarea'

  const [pesquisaAtiva, setPesquisaAtiva] = useState(
    Boolean((tipoTextArea && valor) || (tipoPesquisa && valor)),
  )

  const [value, setValue] = useState(valor?.toString() ?? '')
  const [alturaInicial, setAlturaInicial] = useState<number | undefined>()
  const textAreaRef = useRef<HTMLTextAreaElement>(null)

  const ajustarAltura = useCallback(() => {
    if (textAreaRef.current) {
      const computedStyle = window.getComputedStyle(textAreaRef.current)
      if (!alturaInicial) {
        const lineHeight = isNaN(parseFloat(computedStyle.lineHeight))
          ? undefined
          : parseFloat(computedStyle.lineHeight)

        const paddingTop = parseFloat(computedStyle.paddingTop)
        const paddingBottom = parseFloat(computedStyle.paddingBottom)
        const fontSize = parseFloat(computedStyle.fontSize)
        setAlturaInicial(lineHeight ?? paddingTop + paddingBottom + fontSize)
      }

      const maxAltura =
        numeroMaxLinhas && alturaInicial
          ? numeroMaxLinhas * alturaInicial
          : Infinity

      textAreaRef.current.style.height = 'auto'
      textAreaRef.current.style.height = `${Math.min(textAreaRef.current.scrollHeight, maxAltura)}px`
    }
  }, [numeroMaxLinhas, alturaInicial])

  useEffect(() => {
    if (tipoTextArea && numeroMaxLinhas && textAreaRef.current) {
      const valorTruncado = truncarTexto({
        valor: value,
        numeroMaxLinhas,
        textAreaRef,
      })

      if (valorTruncado !== value) {
        setValue(valorTruncado)
      }
    }
  }, [tipoTextArea, numeroMaxLinhas, value, textAreaRef])

  useEffect(() => {
    setValue(valor?.toString() ?? '')
  }, [valor])

  useEffect(() => {
    ajustarAltura()
  }, [ajustarAltura, value])

  const formatarNumero = (valor: string): string => {
    return valor.replace('.', ',')
  }

  const validarNumero = (numero: string): boolean => {
    return numero === '' || /^\d+(,\d{0,3})?$/.test(numero)
  }

  const handleTipoNumero = (
    valorInput: string,
    tipoNumero: boolean,
    aoDigitar?: (valor: string | number) => void,
  ) => {
    const numeroFormatado = formatarNumero(valorInput)

    if (validarNumero(numeroFormatado)) {
      setValue(numeroFormatado)

      if (aoDigitar) {
        let valorDigitado: string | number
        if (placeholder) {
          valorDigitado = numeroFormatado
        } else {
          valorDigitado = formatarValor(tipoNumero, numeroFormatado)
        }

        aoDigitar(valorDigitado)
      }
    }
  }

  const handleChange = (event: any): void => {
    const valorInput = event.target.value

    if (tipoNumero) {
      handleTipoNumero(valorInput, tipoNumero, aoDigitar)
      return
    }

    setValue(valorInput)

    if (aoDigitar) {
      aoDigitar(valorInput)
    }
  }

  const handleAoDesfocar = () => {
    if (aoDesfocar) {
      aoDesfocar(formatarValor(tipoNumero, value))
    }
  }

  const handleAoFocar = () => {
    if (tipoPesquisa && !value) {
      setPesquisaAtiva(false)
    }
    if (aoFocar) {
      aoFocar()
    }
  }

  const handleSubmeter = (key: any) => {
    if (
      key.type === 'keydown' &&
      !['Enter', 'NumpadEnter'].includes(key.code)
    ) {
      return
    }

    if (aoSubmeter) {
      if (tipoPesquisa) {
        aoSubmeter(value.toUpperCase())
        setPesquisaAtiva(Boolean(value))
      } else {
        aoSubmeter(formatarValor(tipoNumero, value))
      }
    }
  }

  const handleLimparPesquisa = () => {
    if (aoSubmeter) {
      aoSubmeter()
    }

    setValue('')
    setPesquisaAtiva(false)
  }

  return (
    <StyleSheetManager shouldForwardProp={prop => !!prop}>
      <InputContainer
        data-testid={testId}
        onBlur={handleAoDesfocar}
        onFocus={handleAoFocar}
        $corBorda={corBorda ?? cor}
        $corBordaFocada={corBordaFocada}
        $semBorda={semBorda}
        $desabilitado={desabilitado}
        $iconeEsquerda={iconeEsquerda}
        $tipoTextArea={tipoTextArea}
        {...props}
      >
        {tipoTextArea ? (
          <TextAreaStyled
            data-testid={'textarea-' + testId}
            rows={numeroLinhas}
            ref={textAreaRef}
            autoFocus={tipoTextArea}
            value={value}
            onChange={handleChange}
            onKeyDown={handleSubmeter}
            onBlur={handleAoDesfocar}
            cor={textoProps?.cor ?? cor}
            placeholder={placeholder}
            $corTextoDesativado={corTextoDesativado}
            $numeroMaxLinhas={numeroMaxLinhas}
            disabled={desabilitado}
            {...textoProps}
          />
        ) : (
          <InputStyled
            data-testid={'input-' + testId}
            autoFocus={props.autoFocus}
            value={value}
            onChange={handleChange}
            onKeyDown={handleSubmeter}
            onBlur={handleAoDesfocar}
            cor={textoProps?.cor ?? cor}
            placeholder={placeholder}
            $corTextoDesativado={corTextoDesativado}
            disabled={desabilitado}
            {...textoProps}
          />
        )}
        {(tipoPesquisa || tipoTextArea) && !desabilitaIcone && (
          <IconeBotao
            testId={testId}
            icone={icone}
            pesquisaAtiva={pesquisaAtiva}
            itemSelecionado={itemSelecionado}
            handleLimparPesquisa={handleLimparPesquisa}
            handleSubmeter={handleSubmeter}
          />
        )}
      </InputContainer>
    </StyleSheetManager>
  )
}

const IconeBotao = ({
  testId,
  icone,
  pesquisaAtiva,
  itemSelecionado,
  handleLimparPesquisa,
  handleSubmeter,
}: {
  testId: string
  icone?: IconeInput
  pesquisaAtiva: boolean
  itemSelecionado?: boolean
  handleLimparPesquisa: () => void
  handleSubmeter: (key: any) => void
}) => {
  const styleIcone = {
    icone: icone?.icone,
    cor: icone?.cor,
    largura: icone?.largura,
    altura: icone?.altura,
    inclinacao: icone?.inclinacao,
  }

  if (pesquisaAtiva || itemSelecionado) {
    return (
      <InputIconePesquisa
        data-testid={`btn-${testId}-limpar`}
        icone={FecharSVG}
        cor={styleIcone.cor}
        largura={styleIcone.largura}
        altura={styleIcone.altura}
        aoClicar={handleLimparPesquisa}
      />
    )
  }

  return styleIcone.icone ? (
    <Icone
      data-testid={`icone-${testId}-icone`}
      icone={styleIcone.icone}
      cor={styleIcone.cor}
      largura={styleIcone.largura}
      altura={styleIcone.altura}
      inclinacao={styleIcone.inclinacao}
    />
  ) : (
    <InputIconePesquisa
      data-testid={`btn-${testId}-pesquisar`}
      icone={PesquisarSVG}
      cor={styleIcone.cor}
      largura={styleIcone.largura}
      altura={styleIcone.altura}
      aoClicar={handleSubmeter}
    />
  )
}

export const truncarTexto = ({
  valor,
  numeroMaxLinhas,
  textAreaRef,
}: {
  valor: string
  numeroMaxLinhas?: number
  textAreaRef: React.RefObject<HTMLTextAreaElement>
}) => {
  if (valor && textAreaRef.current?.clientWidth && numeroMaxLinhas) {
    const larguraTextArea = textAreaRef.current.clientWidth
    const tamanhoFonte = parseFloat(
      window.getComputedStyle(textAreaRef.current).fontSize,
    )
    const charsPorLinha = Math.floor(larguraTextArea / (tamanhoFonte * 0.64))
    const maxChars = charsPorLinha * numeroMaxLinhas
    if (valor.length > maxChars) {
      return valor.slice(0, maxChars).trim() + '...'
    }
  }

  return valor
}

const formatarValor = (tipoNumero: boolean, valor: string) => {
  if (tipoNumero) {
    const numeroFloat = parseFloat(valor.replace(/\./g, '').replace(',', '.'))
    return isNaN(numeroFloat) ? 0 : numeroFloat
  }

  return valor
}
