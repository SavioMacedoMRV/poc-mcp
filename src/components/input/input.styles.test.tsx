import {render} from '@testing-library/react'
import * as S from './input.styles'

describe('Input.styles', () => {
  test('InputContainer DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.InputContainer />)
    expect(container).toMatchSnapshot()
  })

  test('InputContainer DEVE ser igual ao snapshot QUANDO $tipoTextArea for true', () => {
    const {container} = render(<S.InputContainer $tipoTextArea={true} />)
    expect(container).toMatchSnapshot()
  })

  test('InputContainer DEVE ser igual ao snapshot QUANDO altura for 16', () => {
    const {container} = render(<S.InputContainer altura={16} />)
    expect(container).toMatchSnapshot()
  })

  test('InputContainer DEVE ser igual ao snapshot QUANDO $semBorda for true', () => {
    const {container} = render(<S.InputContainer $semBorda={true} />)
    expect(container).toMatchSnapshot()
  })

  test('InputContainer DEVE ser igual ao snapshot QUANDO $desabilitado for true', () => {
    const {container} = render(<S.InputContainer $desabilitado={true} />)
    expect(container).toMatchSnapshot()
  })

  test('InputContainer DEVE ser igual ao snapshot QUANDO for passado uma cor na $corBordaFocada', () => {
    const {container} = render(<S.InputContainer $corBordaFocada="#00ff00" />)
    expect(container).toMatchSnapshot()
  })

  test('InputStyled DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.InputStyled />)
    expect(container).toMatchSnapshot()
  })

  test('TextAreaStyled DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.TextAreaStyled />)
    expect(container).toMatchSnapshot()
  })

  test('TextAreaStyled DEVE ser igual ao snapshot QUANDO passar o estilo, alinhamento, cursor, transformar, $numeroMaxLinhas e $corTextoDesativado', () => {
    const {container} = render(
      <S.TextAreaStyled
        estilo="bold"
        alinhamento="justify"
        cursor="default"
        transformar="capitalize"
        $numeroMaxLinhas={3}
        $corTextoDesativado="#ff0000"
      />,
    )
    expect(container).toMatchSnapshot()
  })

  test('InputIconePesquisa DEVE ser igual ao snapshot', () => {
    const {container} = render(
      <S.InputIconePesquisa aoClicar={jest.fn()} icone={() => <svg />} />,
    )

    expect(container).toMatchSnapshot()
  })
})
