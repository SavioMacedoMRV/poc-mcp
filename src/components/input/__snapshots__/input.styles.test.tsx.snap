// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.styles InputContainer DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  padding: 8px;
  gap: 8px;
  flex-direction: row;
  justify-content: space-between;
  height: auto;
  width: auto;
  background: #FFFFFF;
  border-radius: 8px;
  border: 2px solid #5D615F;
}

.c0:focus-within {
  border: 2px solid #5D615F;
  box-shadow: 0px 2px 8px 0px rgba(67, 70, 69, 0.3);
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Input.styles InputContainer DEVE ser igual ao snapshot QUANDO $desabilitado for true 1`] = `
.c0 {
  display: flex;
  padding: 8px;
  gap: 8px;
  flex-direction: row;
  justify-content: space-between;
  height: auto;
  width: auto;
  background: #EDF2F0;
  border-radius: 8px;
  border: 2px solid #EDF2F0;
}

.c0:focus-within {
  border: 2px solid #5D615F;
  box-shadow: 0px 2px 8px 0px rgba(67, 70, 69, 0.3);
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Input.styles InputContainer DEVE ser igual ao snapshot QUANDO $semBorda for true 1`] = `
.c0 {
  display: flex;
  padding: 8px;
  gap: 8px;
  flex-direction: row;
  justify-content: space-between;
  height: auto;
  width: auto;
  background: #FFFFFF;
  border-radius: 8px;
  border: none;
}

.c0:focus-within {
  border: none;
  box-shadow: none;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Input.styles InputContainer DEVE ser igual ao snapshot QUANDO $tipoTextArea for true 1`] = `
.c0 {
  display: flex;
  padding: 4px 0;
  gap: 8px;
  flex-direction: row;
  justify-content: space-between;
  height: auto;
  width: auto;
  background: #FFFFFF;
  border-radius: 8px;
  border: 2px solid #5D615F;
}

.c0:focus-within {
  border: 2px solid #5D615F;
  box-shadow: 0px 2px 8px 0px rgba(67, 70, 69, 0.3);
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Input.styles InputContainer DEVE ser igual ao snapshot QUANDO altura for 16 1`] = `
.c0 {
  display: flex;
  padding: 8px;
  gap: 8px;
  flex-direction: row;
  justify-content: space-between;
  height: 16px;
  width: auto;
  background: #FFFFFF;
  border-radius: 8px;
  border: 2px solid #5D615F;
}

.c0:focus-within {
  border: 2px solid #5D615F;
  box-shadow: 0px 2px 8px 0px rgba(67, 70, 69, 0.3);
}

<div>
  <div
    altura="16"
    class="c0"
  />
</div>
`;

exports[`Input.styles InputContainer DEVE ser igual ao snapshot QUANDO for passado uma cor na $corBordaFocada 1`] = `
.c0 {
  display: flex;
  padding: 8px;
  gap: 8px;
  flex-direction: row;
  justify-content: space-between;
  height: auto;
  width: auto;
  background: #FFFFFF;
  border-radius: 8px;
  border: 2px solid #5D615F;
}

.c0:focus-within {
  border: 2px solid #00ff00;
  box-shadow: 0px 2px 8px 0px rgba(67, 70, 69, 0.3);
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Input.styles InputIconePesquisa DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  border-radius: 8px;
  border: none;
  align-items: center;
  justify-content: center;
  background: transparent;
  padding: 0px;
  opacity: 1;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.7;
}

.c0:active {
  opacity: 1;
}

.c0:focus-visible {
  outline: unset;
  border: 2px solid #999E9C;
}

.c1 {
  align-items: baseline;
}

<div>
  <button
    class="c0 c1"
    cor="#434645"
  >
    <svg />
  </button>
</div>
`;

exports[`Input.styles InputStyled DEVE ser igual ao snapshot 1`] = `
.c0 {
  border: none;
  outline: none;
  background: transparent;
  width: 100%;
  font-size: 14px;
  line-height: normal;
  font-family: AvertaStd-Semibold;
  color: #242625;
  text-align: start;
  cursor: text;
  text-transform: none;
}

.c0:focus::placeholder {
  color: transparent;
}

.c0:disabled {
  color: #999E9C;
  cursor: default;
}

.c0::-webkit-search-cancel-button {
  -webkit-appearance: none;
  background: #999E9C;
  height: 10px;
  width: 10px;
  border-radius: 5px;
}

.c0::-webkit-inner-spin-button {
  -webkit-appearance: none;
}

.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
}

<div>
  <input
    class="c0"
  />
</div>
`;

exports[`Input.styles TextAreaStyled DEVE ser igual ao snapshot 1`] = `
.c0 {
  border: none;
  outline: none;
  background: transparent;
  width: 100%;
  font-size: 14px;
  line-height: normal;
  font-family: AvertaStd-Semibold;
  color: #242625;
  text-align: start;
  cursor: text;
  text-transform: none;
  resize: none;
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  white-space: normal;
  box-sizing: border-box;
}

.c0:focus::placeholder {
  color: transparent;
}

.c0:disabled {
  color: #999E9C;
  cursor: default;
}

.c0::-webkit-search-cancel-button {
  -webkit-appearance: none;
  background: #999E9C;
  height: 10px;
  width: 10px;
  border-radius: 5px;
}

.c0::-webkit-inner-spin-button {
  -webkit-appearance: none;
}

.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
}

<div>
  <textarea
    class="c0"
  />
</div>
`;

exports[`Input.styles TextAreaStyled DEVE ser igual ao snapshot QUANDO passar o estilo, alinhamento, cursor, transformar, $numeroMaxLinhas e $corTextoDesativado 1`] = `
.c0 {
  border: none;
  outline: none;
  background: transparent;
  width: 100%;
  font-size: 14px;
  line-height: normal;
  font-family: AvertaStd-Black;
  color: #242625;
  text-align: justify;
  cursor: default;
  text-transform: capitalize;
  resize: none;
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  white-space: normal;
  box-sizing: border-box;
}

.c0:focus::placeholder {
  color: transparent;
}

.c0:disabled {
  color: #ff0000;
  cursor: default;
}

.c0::-webkit-search-cancel-button {
  -webkit-appearance: none;
  background: #999E9C;
  height: 10px;
  width: 10px;
  border-radius: 5px;
}

.c0::-webkit-inner-spin-button {
  -webkit-appearance: none;
}

.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
}

<div>
  <textarea
    alinhamento="justify"
    class="c0"
    cursor="default"
    estilo="bold"
    transformar="capitalize"
  />
</div>
`;
